serial/uart.c

```diff
@@ -37,6 +38,56 @@ static UartRecv_cb uart_recv_cb[4];
 static void uart0_rx_intr_handler(void *para);
 
 /******************************************************************************
+ * FunctionName : set_tx_enable_pin
+ * Description  : Set which pin to use for RS-485 TX_ENABLE
+ * Parameters   : pin, the pin to use
+ * Returns      : NONE
+*******************************************************************************/
+void ICACHE_FLASH_ATTR
+uart0_set_tx_enable_pin(int8_t pin) {
+	uart0_tx_enable_pin = pin;
+}
+
+/******************************************************************************
+ * FunctionName : tx_enable
+ * Description  : Internal used function
+ *                Set the TX_ENABLE line for RS-485 communications
+ * Parameters   : state, true if the TX_ENABLE line should be asserted high
+ * Returns      : NONE
+*******************************************************************************/
+static void ICACHE_FLASH_ATTR
+tx_enable(bool state)
+{
+  if (uart0_tx_enable_pin >= 0) {
+#ifdef SERBR_DBG
+    os_printf("TX_ENABLE gpio%d state=%d\n", uart0_tx_enable_pin, (int)state);
+#endif
+    GPIO_OUTPUT_SET(uart0_tx_enable_pin, (state) ? 1 : 0);
+  }
+#ifdef SERBR_DBG
+  else { os_printf("TX Enable: no pin\n"); }
+#endif
+}
+
+os_timer_t uart_tx_enable_timer;
+bool uart_tx_enable_timer_inited = false;
+
+/******************************************************************************
+ * FunctionName : tx_completed_interrupt
+ * Description  : Internal used function
+ *                Set the TX enable line low, after the UART has completed tranmission
+ * Parameters   : unused unused
+ * Returns      : NONE
+*******************************************************************************/
+static void ICACHE_FLASH_ATTR
+tx_completed_interrupt(void *unused)
```

Comment:
Something is wrong witch this interrupt. If esp start transmit newer come back to listening.